src/Raven.Server/Documents/Queries/Graph/EdgeQueryStep.cs

```diff
@@ -0,0 +1,143 @@
+                                    {
+                                        var doc = right.GetSingleDocumentResult(alias);
+                                        if (doc.Id != kvp.Key)
+                                            continue;
+
+                                        var clone = new Match(left);
+                                        clone.Merge(right);
+                                        //clone -> result row
+                                        res.Add(clone);
+                                        break;
+                                    }
+                                }
+                            }
+                        }
+
+                        break;
+                    case BlittableJsonReaderArray array:
+                        break;
+                    case BlittableJsonReaderObject json:
+                        if (edge.Where.IsMatchedBy(json, _queryParameters))
+                        {
+                            //hasResults 
+                            var includeFields = new EdgeIncludeOp();
+                            IncludeUtil.GetDocIdFromInclude(json, edge.Path.FieldValue, includeFields);
+                            if (_included.Count > 0)
+                            {
+                                foreach (var kvp in _included)
+                                {
+                                    if (kvp.Key == null)
+                                        continue;
+
+                                    foreach (var right in rightResult)
+                                    {
+                                        foreach (var alias in right.Aliases)
+                                        {
+                                            var doc = right.GetSingleDocumentResult(alias);
+                                            if (doc.Id != kvp.Key)
+                                                continue;
+
+                                            var clone = new Match(left);
+                                            clone.Merge(right);
+                                            //clone -> result row
+                                            res.Add(clone);
+                                            break;
+                                        }
+                                    }
+                                }
+                            }
+
+                        }
+                        break;
+                }
+            }
+
+            return res;
+        }
+
+        private IQueryStep _left;
+        private IQueryStep _right;
+        private MatchPath _edgePath;
+        private readonly BlittableJsonReaderObject _queryParameters;
+        private WithEdgesExpression _edgesExpression;
+        private Dictionary<string, BlittableJsonReaderObject> _included = new Dictionary<string, BlittableJsonReaderObject>();
+
+        public IEnumerable<IQueryStep> Dependencies => GetDependencies();
```

Comment:
Why do you care about this?

Reply:
At first, the query plan was executing the steps but I figured it makes no sense.
I left it thinking that the optimization might care about it but it is probably redundant and will be removed.